eZ/Publish/Core/Search/Legacy/Content/WordIndexer/Repository/SearchIndex.php

```diff
@@ -1,266 +1,264 @@
-     *
-     * @return array
      */
-    public function getWords(array $words)
+    public function getWords(array $words): array
     {
-        $query = $this->dbHandler->createSelectQuery();
+        $query = $this->connection->createQueryBuilder();
 
-        // use array_map as some DBMS-es do not cast integers to strings by default
-        $query->select('*')
-            ->from('ezsearch_word')
-            ->where($query->expr->in('word', array_map('strval', $words)));
+        $query
+            ->select('*')
+            ->from(self::SEARCH_WORD_TABLE)
+            ->where($query->expr()->in('word', ':words'))
+            // use array_map as some DBMS-es do not cast integers to strings by default
+            ->setParameter('words', array_map('strval', $words), Connection::PARAM_STR_ARRAY);
 
-        $stmt = $query->prepare();
-        $stmt->execute();
-
-        return $stmt->fetchAll(PDO::FETCH_ASSOC);
+        return $query->execute()->fetchAll(FetchMode::ASSOCIATIVE);
     }
 
     /**
      * Increase the object count of the given words by one.
      *
      * @param array $wordId
      */
-    public function incrementWordObjectCount(array $wordId)
+    public function incrementWordObjectCount(array $wordId): void
     {
-        $this->updateWordObjectCount($wordId, ['object_count' => 'object_count + 1']);
+        $this
+            ->getWordUpdateQuery($wordId)
+            ->set('object_count', 'object_count + 1')
+            ->execute();
     }
 
     /**
      * Decrease the object count of the given words by one.
      *
      * @param array $wordId
      */
-    public function decrementWordObjectCount(array $wordId)
+    public function decrementWordObjectCount(array $wordId): void
     {
-        $this->updateWordObjectCount($wordId, ['object_count' => 'object_count - 1']);
+        $this
+            ->getWordUpdateQuery($wordId)
+            ->set('object_count', 'object_count - 1')
+            ->execute();
     }
 
     /**
      * Insert new words (legacy db table: ezsearch_word).
      *
      * @param array $words
      */
-    public function addWords(array $words)
+    public function addWords(array $words): void
     {
-        $query = $this->dbHandler->createInsertQuery();
-        $query->insertInto('ezsearch_word');
-
-        $word = null;
+        $query = $this->connection->createQueryBuilder();
+        $query
+            ->insert(self::SEARCH_WORD_TABLE)
+            ->values(
+                [
+                    'word' => ':word',
+                    'object_count' => ':one',
+                ]
+            )
+            ->setParameter(':one', 1, ParameterType::INTEGER);
 
-        $query->set(
-            'word',
-            ':word'
-        )->set(
-            'object_count',
-            '1'
-        );
-        $stmt = $query->prepare();
         foreach ($words as $word) {
-            $stmt->execute(['word' => $word]);
+            $query->setParameter('word', $word);
```

Comment:
There is a little bit inconsistency here. 

```suggestion
            $query->setParameter(':word', $word);
```

Reply:
> There is a little bit inconsistency here.

Actually... no :)
Parameters when bound (or rather used in a query) should have a leading colon. When set - should not. They're gonna work, but require more operations, see [extractParam](https://github.com/doctrine/dbal/blob/v2.10.1/lib/Doctrine/DBAL/SQLParserUtils.php#L272-L279) and [createNamedParameter](https://github.com/doctrine/dbal/blob/v2.10.1/lib/Doctrine/DBAL/Query/QueryBuilder.php#L1278). Though doc for `setParameter` still refers to the "old" syntax, weird. I'd keep it as is due to the implementation of `extractParam`.